hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/ClusterMapReduceTestCase.java

```diff
@@ -43,8 +45,21 @@
  * The DFS filesystem is formated before the testcase starts and after it ends.
  */
 public abstract class ClusterMapReduceTestCase {
+  private static final String TEST_ROOT_DEFAULT_PATH =
+      System.getProperty("test.build.data", "target/test-dir");
+  private static Path testRootDir;
+
   private MiniDFSCluster dfsCluster = null;
-  private MiniMRCluster mrCluster = null;
+  private MiniMRClientCluster mrCluster = null;
+
+  protected static void setupClassBase(Class<?> testClass) throws Exception {
+    // setup the test root directory
+    testRootDir = GenericTestUtils.setupTestRootDir(testClass,
+        TEST_ROOT_DEFAULT_PATH);
+    System.setProperty(GenericTestUtils.SYSPROP_TEST_DATA_DIR,
```

Comment:
I don't think we should set the property here if we are doing it in setupTestRootDir().

Reply:
One question is whether we should get the original value and then reset it in an afterClass method?  Is every unit test run in a separate jvm instance?   Wouldn't want a following test to get the wrong setting.

Reply:
It will be set for the JVM executing the unittest. Each unit test should then have its own property. Then they should not overlap.